lib/modules/manager/maven-wrapper/artifacts.ts

```diff
@@ -0,0 +1,200 @@
+  if (status.modified.includes(fileProjectPath)) {
+    return {
+      file: {
+        type: 'addition',
+        path: fileProjectPath,
+        contents: await readLocalFile(fileProjectPath),
+      },
+    };
+  }
+  return null;
+}
+
+export async function updateArtifacts({
+  packageFileName,
+  newPackageFileContent,
+  updatedDeps,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  try {
+    logger.debug({ updatedDeps }, 'maven-wrapper.updateArtifacts()');
+
+    if (
+      !updatedDeps.some(
+        (dep) => dep.depName === 'org.apache.maven.wrapper:maven-wrapper'
+      )
+    ) {
+      logger.info(
+        'Maven wrapper version not updated - skipping Artifacts update'
+      );
+      return null;
+    }
+
+    const cmd = await createWrapperCommand();
+    if (!cmd) {
+      logger.info('No mvnw found - skipping Artifacts update');
+      return null;
+    }
+    await executeWrapperCommand(cmd, config);
+
+    const status = await getRepoStatus();
+    const artifactFileNames = [
+      '.mvn/wrapper/maven-wrapper.properties',
+      '.mvn/wrapper/maven-wrapper.jar',
+      '.mvn/wrapper/MavenWrapperDownloader.java',
+      'mvnw',
+      'mvnw.cmd',
+    ].map(
+      (filename) =>
+        packageFileName.replace('.mvn/wrapper/maven-wrapper.properties', '') +
+        filename
+    );
+    const updateArtifactsResult = (
+      await getUpdatedArtifacts(status, artifactFileNames)
+    ).filter(is.truthy);
+
+    logger.debug(
+      { files: updateArtifactsResult.map((r) => r.file?.path) },
+      `Returning updated maven-wrapper files`
+    );
+    return updateArtifactsResult;
+  } catch (err) {
+    logger.debug({ err }, 'Error setting new Maven Wrapper release value');
+    return [
+      {
+        artifactError: {
+          lockFile: packageFileName,
+          stderr: err.message,
+        },
+      },
+    ];
+  }
+}
+
+async function getUpdatedArtifacts(
+  status: StatusResult,
+  artifactFileNames: string[]
+): Promise<(UpdateArtifactsResult | null)[]> {
+  const updatedResults: (UpdateArtifactsResult | null)[] = [];
+  for (const artifactFileName of artifactFileNames) {
+    updatedResults.push(await addIfUpdated(status, artifactFileName));
+  }
+  return updatedResults;
+}
+
+/**
+ * Find compatible java version for maven.
+ * see https://maven.apache.org/developers/compatibility-plan.html
+ * @param mavenWrapperVersion current maven version
+ * @returns A Java semver range
+ */
+export function getJavaConstraint(
+  mavenWrapperVersion: string | null | undefined
+): string | null {
+  const major = mavenWrapperVersion
+    ? mavenVersioning.getMajor(mavenWrapperVersion)
+    : null;
+
+  if (major && major >= 3) {
+    return '^17.0.0';
+  }
+
+  return '^8.0.0';
+}
+
+async function executeWrapperCommand(
+  cmd: string,
+  config: UpdateArtifactsConfig
+): Promise<void> {
+  logger.debug(`Updating maven wrapper: "${cmd}"`);
+  const execOptions: ExecOptions = {
```

Comment:
add cwdFile so we are in right folder for execute

Reply:
nope, still missing. see
https://github.com/renovatebot/renovate/blob/07460542850cc79e934a8d4606371c88776e4c0c/lib/modules/manager/gradle-wrapper/artifacts.ts#L149